ocaml/networkd/lib/network_utils.ml

```diff
@@ -184,14 +184,27 @@ module Sysfs = struct
   let is_physical name =
     try
       let devpath = getpath name "device" in
-      let driver_link = Unix.readlink (devpath ^ "/driver") in
-      (* filter out symlinks under device/driver which look like
-         /../../../devices/xen-backend/vif- *)
-      not
-        (List.mem "xen-backend"
-           (Astring.String.cuts ~empty:false ~sep:"/" driver_link)
-        )
-    with _ -> false
+      let is_vif =
+        let driver_link = Unix.readlink (devpath ^ "/driver") in
+        (* filter out symlinks under device/driver which look like
+           /../../../devices/xen-backend/vif- *)
+        List.mem "xen-backend"
+          (Astring.String.cuts ~empty:false ~sep:"/" driver_link)
+      in
+      let is_vf =
+        try
+          ignore @@ Unix.readlink (devpath ^ "/physfn") ;
+          true
+        with _ -> false
+      in
+      let is_phys = not (is_vif || is_vf) in
+      if not is_phys then
+        debug "%s is not physical: is_vif=%B, is_vf=%B" name is_vif is_vf ;
```

Comment:
Is this needed, I'm afraid some callers my call the function in high frequency, adding a lot of loglines which are not useful.